beetsplug/lyrics.py

```diff
@@ -695,40 +700,47 @@ def func(lib, opts, args):
                     if opts.printlyr:
                         ui.print_(item.lyrics)
                     if opts.writerst:
-                        if artist != item.artist:
-                            artist = item.artist
-                            if output != sys.stdout:
-                                output.close()
-                            slug = re.sub(r'\W+', '-',
-                                          unidecode(artist).lower())
-                            path = os.path.join(opts.writerst, slug + u'.rst')
-                            output = open(path, 'w')
-                            rst = u'''%s
+                        self.writerst(opts.writerst, item)
+            if opts.writerst:
+                # flush last artist
+                self.writerst(opts.writerst, None)
+        cmd.func = func
+        return [cmd]
+
+    def writerst(self, directory, item):
+        """Write the item to an RST file
+
+        this will keep state (in the `rst` variable) in order to avoid
+        writing continuously to the same files
+        """
+        if item is None or self.artist != item.artist:
+            if self.rst is not None:
+                slug = re.sub(r'\W+', '-', unidecode(self.artist).lower())
+                path = os.path.join(directory, 'artists', slug + u'.rst')
+                with open(path, 'w') as output:
```

Comment:
ugh. typical python crap. let me guess: you're using Python 3?

isn't the `encode('utf-8')` idiom okay anymore?